gnucash/import-export/import-backend.c

```diff
@@ -827,6 +839,90 @@ void split_find_match (GNCImportTransInfo * trans_info,
 /***********************************************************************
  */
 
+/* append the imported transaction description to the matched transaction description */
+static void
+desc_append (const GNCImportTransInfo* trans_info)
+{
+    GNCImportMatchInfo *selected_match =
+                gnc_import_TransInfo_get_selected_match (trans_info);
+
+    gchar* tmp = g_strconcat( xaccTransGetDescription (selected_match->trans),
+                              "|",
+                              xaccTransGetDescription (gnc_import_TransInfo_get_trans( trans_info)),
+                              NULL);
+    xaccTransSetDescription (selected_match->trans, tmp);
+    g_free (tmp);
+}
+
+/* replace the matched transaction description with the imported transaction description */
+static void
+desc_replace (const GNCImportTransInfo* trans_info)
+{
+    GNCImportMatchInfo *selected_match =
+                gnc_import_TransInfo_get_selected_match (trans_info);
+
+    xaccTransSetDescription (selected_match->trans,
+                             xaccTransGetDescription (
+                                gnc_import_TransInfo_get_trans (trans_info)));
+}
+
+/* append the imported transaction notes to the matched transaction notes */
+static void
+notes_append (const GNCImportTransInfo* trans_info)
+{
+    GNCImportMatchInfo *selected_match =
+                gnc_import_TransInfo_get_selected_match (trans_info);
+
+    gchar* tmp = g_strconcat (xaccTransGetNotes (selected_match->trans),
+                              "|",
+                              xaccTransGetNotes (gnc_import_TransInfo_get_trans (trans_info)),
+                              NULL);
+    xaccTransSetNotes (selected_match->trans, tmp);
+    g_free (tmp);
+}
+
+/* replace the matched transaction notes with the imported transaction notes */
+static void
+notes_replace (const GNCImportTransInfo* trans_info)
+{
+    GNCImportMatchInfo *selected_match =
+                gnc_import_TransInfo_get_selected_match(trans_info);
+
+    xaccTransSetNotes(selected_match->trans,
+                      xaccTransGetNotes(
+                          gnc_import_TransInfo_get_trans(trans_info)));
+}
+
+/* Append or replace transaction description and notes
+ * depending on the Append checkbox
+ */
+static void
+update_desc_and_notes (const GNCImportTransInfo* trans_info)
+{
+    GNCImportMatchInfo* selected_match =
+            gnc_import_TransInfo_get_selected_match (trans_info);
+
```

Comment:
Why not add...
`Transaction* trans = gnc_import_TransInfo_get_trans (trans_info);`
here and then substitute where used below. Also you could add 'selected_match' and 'trans' as parameters to the four functions like `notes_replace (trans_info, selected_match, trans)` that way you only be retrieving those parameters once.